src/src/com/microsoft/aad/adal/SerializedTokenCacheItem.java

```diff
@@ -0,0 +1,42 @@
+package com.microsoft.aad.adal;
+
+import com.google.gson.annotations.SerializedName;
+
+final class SerializedTokenCacheItem {
+
+    private static final String TAG = SerializedTokenCacheItem.class.getSimpleName();
+
+    @SerializedName("authority")
+    private final String mAuthority;
+
+    @SerializedName("refreshToken")
+    private final String mRefreshToken;
+
+    @SerializedName("idToken")
+    private final String mIdToken;
+
+    @SerializedName("familyClientId")
+    private final String mFamilyClientId;
+
+    SerializedTokenCacheItem(final TokenCacheItem tokenCacheItem) {
+        this.mAuthority = tokenCacheItem.getAuthority();
+        this.mRefreshToken = tokenCacheItem.getRefreshToken();
+        this.mIdToken = tokenCacheItem.getRawIdToken();
+        this.mFamilyClientId = tokenCacheItem.getFamilyClientId();
+    }
+
+    TokenCacheItem parseSerializedTokenCacheItem() throws AuthenticationException {
+        TokenCacheItem tokenCacheItem = new TokenCacheItem();
+        IdToken idToken;
```

Comment:
These should not be deleted. Because this.mIdToken is just a string and we need to construct it to a idToken object.